Example/WalletApp/PresentationLayer/Wallet/Main/MainPresenter.swift

```diff
@@ -71,8 +84,59 @@ extension MainPresenter {
                     AlertPresenter.present(message: "No common chains", type: .error)
                     return
                 }
+
                 router.present(request: result.request, importAccount: importAccount, context: result.context)
             }
             .store(in: &disposeBag)
     }
+
+    private func tryRoutCATransaction(request: Request, context: VerifyContext?) async throws {
+        guard request.method == "eth_sendTransaction" else {
+            return
+        }
+        do {
+            let tx = try request.params.get([Tx].self)[0]
+
+            let transaction = InitialTransaction(
+                chainId: request.chainId.absoluteString,
+                from: tx.from,
+                to: tx.to,
+                value: "0",
+                input: tx.data
+            )
+
+
+            ActivityIndicatorManager.shared.start()
+            let routeResponseSuccess = try await WalletKit.instance.prepare(transaction: transaction)
+
+            await MainActor.run {
+                switch routeResponseSuccess {
+                case .success(let routeResponseSuccess):
+                    switch routeResponseSuccess {
+                    case .available(let routeResponseAvailable):
+                        router.presentCATransaction(sessionRequest: request, importAccount: importAccount, routeResponseAvailable: routeResponseAvailable, context: context)
+                    case .notRequired(let routeResponseNotRequired):
+                        AlertPresenter.present(message: "Routing not required", type: .success)
+                        router.present(sessionRequest: request, importAccount: importAccount, sessionContext: context)
+                    }
+                case .error(let routeResponseError):
+                    AlertPresenter.present(message: "Route response error: \(routeResponseError)", type: .success)
+                    Task {
+                        try await WalletKit.instance.respond(
+                            topic: request.topic,
+                            requestId: request.id,
+                            response: .error(.init(code: 0, message: ""))
+                        )
+                    }
+                }
+            }
+            ActivityIndicatorManager.shared.stop()
+        } catch {
```

Comment:
If there is a CA error, does the wallet still try to process the request as-normal?

Reply:
currently wallet does not handle non-CA eth_sendTransaction